src/async.rs

```diff
@@ -86,17 +87,18 @@ impl<'d> Transfer<'d> {
     }
 
     /// Access the buffer of a transfer.
-    pub fn buffer(&mut self) -> &'d mut [u8] {
-        unsafe { slice::from_raw_parts_mut((*self.transfer).buffer, (*self.transfer).length as usize) }
+    pub fn buffer(&mut self) -> &mut Vec<u8> {
+        &mut self.buffer
     }
 
     /// Replace the buffer of a transfer.
-    pub fn set_buffer(&mut self, buffer: &'d mut [u8]) {
+    pub fn set_buffer(&mut self, mut buffer: Vec<u8>) {
         unsafe {
             (*self.transfer).buffer = buffer.as_mut_ptr();
```

Comment:
This aliases the mutable memory in the vector and is unsafe. If you use the reference returned by `buffer()` to resize the Vec, it invalidates this pointer.